### PR TITLE
Xorshift* generator

### PR DESCRIPTION
**Pull request by ghost**
My first time writing any Ruby (I'm a Python fan myself), so be gentle. I implemented a 64-bit xorshift\* PRNG (took me a while to find a PRNG I could actually code that wasn't already implemented), which does run with ./bin/entrogen xorshiftstar -l 10485760 | entrospect and appears to produce fairly decent random numbers.

### CODE REVIEW

File: lib/generators/xorshift.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  def xorshift_star(sseed)
+    x = sseed.unpack('Q>')[0]
+    x ^= x >> 12
+    x ^= x << 25
+    x ^= x >> 27
+    return x * 2685821657736338717
+  end
+    
+  def next_chunk
+    @i = xorshift_star([@i].pack('Q>'))
```
Comment (kedji) [inline lib/generators/xorshift.rb]: It looks like you're packing `@i` just to unpack it in `xorshift_star` to keep `@i` from getting too large. I think it'd be much more efficient to mod or mask bits off the return value on line 34.

**Comment by kedji** [comment]
This is a good generator. I only have one suggestion about your code.

**Comment by ghost** [comment]
Thanks Eric. Yeah I was not super happy about that part, but being new to
Ruby, I wasn't sure how else to do it. The other reason I was doing it was
to try to emulate the C-algorithm, which explicitly used unsigned 64-bit
ints. Not being an expert in cryptography or how that algorithm works, I
figured it would be safer to pack it to a 64-bit uint and then unpack it.
How would you recommend I keep that return from going outside of the 64-bit
uint?

Thanks for the feedback!

Greg

**Comment by kedji** [comment]
`return (x * 2685821657736338717) % 2**64`

**Pull request merged**